core/src/utils.jl

```diff
@@ -63,14 +71,33 @@ function create_graph(db::DB)::MetaGraph
         end
         edge_metadata =
             EdgeMetadata(fid, edge_type, allocation_network_id, id_src, id_dst, false)
+        graph[id_src, id_dst] = edge_metadata
+        flow_counter += 1
+        flow_dict[(id_src, id_dst)] = flow_counter
         if allocation_network_id != 0
             if !haskey(edges_source, allocation_network_id)
                 edges_source[allocation_network_id] = Set{EdgeMetadata}()
             end
             push!(edges_source[allocation_network_id], edge_metadata)
         end
-        graph[id_src, id_dst] = edge_metadata
     end
+
+    flow = zeros(flow_counter)
+    flow_vertical = zeros(flow_vertical_counter)
+    if config.solver.autodiff
+        flow = DiffCache(flow, chunk_size)
+        flow_vertical = DiffCache(flow_vertical, chunk_size)
+    end
```

Comment:
```suggestion
    flow, flow_vertical = if config.solver.autodiff
        DiffCache(flow, chunk_size), DiffCache(flow_vertical, chunk_size)
    else
        zeros(flow_counter), zeros(flow_vertical_counter)
    end
```
It's clearer and saves doesn't allocate needlessly

Reply:
The reason it is written this way is because allocation of `flow` and `flow_vertical` is always needed, and in the case of autodiff `DiffCache` allocates a bit more for AD computations. No needless allocation happens

Reply:
Your solution always allocates the zeros.
Mine only allocates them if `config.solver.autodiff` is false.

Reply:
I've overlooked that flow is an input parameter of `DiffCache`